### PR TITLE
feat: minify and notify clipable email body

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

[Email content larger than 102kb gets clipped by Gmail](https://mailchimp.com/pt-br/help/gmail-is-clipping-my-email/). This PR implements a notice to let the user know their content may get clipped if we detect a resulting body larger than 100,000 bytes. This approximation is assuming that the ESP will add extra tags for tracking, footer, etc.

image

It also implements the minifying options to MJML to reduce the chance of clipping. My local tests produced a reduction of 25% of an email with ~100k bytes.

### How to test the changes in this Pull Request:

1. Check out this PR and run `npm run build`
2. Send an email using multiple different blocks to make sure the markup is not altered by the minification
3. Add multiple text blocks in a newsletter draft until it reaches > 100k bytes
4. Save the draft and confirm you see the notice in the screenshot above

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/newsletter-editor/editor/index.js [hunk 1]
```diff
@@ -133,6 +142,23 @@ const Editor = compose( [
 		}
 	}, [ props.status ] );
 
+	// Notify if email content is larger than ~100kb.
+	useEffect( () => {
+		const noticeId = 'newspack-newsletters-email-content-too-large';
+		const message = __(
+			'Email content is too large and may get clipped by email clients.',
```
Comment (dkoo) [inline src/newsletter-editor/editor/index.js:149]: Nit: "long" sounds to me like it better suggests what action the editor should take to fix this error.

```suggestion
			'Email content is too long and may get clipped by email clients.',
```

**Comment by miguelpeixe** [comment]
> except the new WP 5.9 "Row" block seems to not render correctly

Nice catch! I just pushed #772 for this.

> Inspecting the meta value, it does appear to be minified, but I wonder if the above indicates support for this feature might go away at some point?

I didn't notice that warning, but [unfortunately yes](https://github.com/mjmlio/mjml/pull/2204) ☹️ 
We could use the minify option until the next stable major release is out and until then figure out if we'll use a different library for the task. WDYT?

**Review by dkoo — APPROVED** [review]
>  We could use the minify option until the next stable major release is out and until then figure out if we'll use a different library for the task. We can also not minify at all and just use the notice. WDYT?

This sounds good to me, as long as continuing to use the `minify` option doesn't end up causing an error on the MJML side. I'm assuming after the option is removed from the library behavior will revert back to what's happening on the current `master` HEAD, right? Basically the meta value will be saved as unminified HTML again?

**Comment by dkoo** [comment]
That matches my understanding, too—thanks!

**Pull request merged**